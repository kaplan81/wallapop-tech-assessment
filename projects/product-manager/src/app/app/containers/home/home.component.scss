.home {
  width: 960px;
  margin: 60px auto;
  &__heading {
    font-size: 2rem;
    font-weight: bold;
    text-align: center;
    margin-bottom: 40px;
  }
  &__paragraph {
    font-size: 1.1rem;
    text-align: center;
  }
}
