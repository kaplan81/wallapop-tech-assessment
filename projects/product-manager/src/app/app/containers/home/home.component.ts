import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  changeDetection: ChangeDetectionStrategy.OnPush,
  imports: [],
  selector: 'mng-home',
  standalone: true,
  styleUrls: ['./home.component.scss'],
  templateUrl: './home.component.html',
})
export class HomeComponent {}
