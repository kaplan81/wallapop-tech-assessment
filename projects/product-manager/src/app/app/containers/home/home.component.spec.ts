import { ComponentFixture, TestBed } from '@angular/core/testing';
import { HomeComponent } from './home.component';

describe('HomeComponent', () => {
  let fixture: ComponentFixture<HomeComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [HomeComponent],
    }).compileComponents();
    fixture = TestBed.createComponent(HomeComponent);
  });

  it('should match snapshot', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
