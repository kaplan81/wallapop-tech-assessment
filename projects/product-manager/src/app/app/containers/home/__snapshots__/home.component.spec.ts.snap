// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should match snapshot 1`] = `
<mng-home
  __ngContext__="0"
>
  <div
    class="home"
  >
    <h1
      class="home__heading"
    >
      Welcome to our Product Manager
    </h1>
    <p
      class="home__paragraph"
    >
      Please start by perfoming a wonderful fake search.
    </p>
    <br />
    <p
      class="home__paragraph"
    >
      The API only returns 20 items...
    </p>
    <br />
    <p
      class="home__paragraph"
    >
      ...and they are all totally different from each other.
    </p>
    <br />
    <p
      class="home__paragraph"
    >
      Meaning any actual search would return only 1 element.
    </p>
    <br />
    <p
      class="home__paragraph"
    >
       We prefer to peform a fake search to be able to sort the results later. 
    </p>
    <br />
    <p
      class="home__paragraph"
    >
      So just type anything and hit Enter.
    </p>
    <br />
    <p
      class="home__paragraph"
    >
      Have fun!
    </p>
  </div>
</mng-home>
`;
